cluster/service/cluster_whitebox_test.go

```diff
@@ -0,0 +1,165 @@
+package service
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/test/recorder"
+	testsuite "github.com/fabric8-services/fabric8-auth/test/suite"
+	tokentestsupport "github.com/fabric8-services/fabric8-auth/test/token"
+	"github.com/fabric8-services/fabric8-auth/token"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestCluster(t *testing.T) {
+	suite.Run(t, &TestClusterSuite{})
+}
+
+type TestClusterSuite struct {
+	testsuite.UnitTestSuite
+	cs      service.ClusterService
+	tm      token.Manager
+	saToken string
+}
+
+func (s *TestClusterSuite) SetupSuite() {
+	s.UnitTestSuite.SetupSuite()
+	var err error
+	s.tm, err = token.DefaultManager(s.Config)
+	require.NoError(s.T(), err)
+	s.saToken = s.tm.AuthServiceAccountToken()
+}
+
+func (s *TestClusterSuite) SetupTest() {
+	clusterCache = nil
+	started = 0
+	s.cs = NewClusterService(nil, s.Config)
+}
+
+func (s *TestClusterSuite) TearDownTest() {
+	if clusterCache != nil {
+		clusterCache.stop()
+	}
+}
+
+func (s *TestClusterSuite) TestClustersFail() {
+	ctx, _, reqID := tokentestsupport.ContextWithTokenAndRequestID(s.T())
+
+	s.T().Run("clusters() fails if can't get clusters", func(t *testing.T) {
```

Comment:
OH WAIT! Did you use subtests here ?!? 😂

Reply:
I was thinking about you when adding them ;)
Just to clarify my position. It's OK to use subtests when you don't need to sacrifice automated test setup/cleanup to achieve proper test isolation.

In this case I actually needed subtests to depend on each other. I need the second subtest to be executed after the first one.
The alternative of using subtests would be to have one big test function TestClustersFail() and merge both subtests there. It's the case when I do not want subtests to be isolated. Subtests here is just to represent block of codes/tests more nicely.